verifier/verifier.go

```diff
@@ -518,6 +545,67 @@ func verifyAuthenticTimestamp(outcome *notation.VerificationOutcome) *notation.V
 	}
 }
 
+func verifyRevocation(outcome *notation.VerificationOutcome, client *http.Client, logger log.Logger) *notation.ValidationResult {
+	r, err := revocation.New(client)
+	if err != nil {
+		logger.Debug("no HTTP client provided for revocation checking")
+		return &notation.ValidationResult{
+			Type:   trustpolicy.TypeRevocation,
+			Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+			Error:  errors.New("unable to check revocation status, an HTTP client must be provided"),
+		}
+	}
+
+	authenticSigningTime := outcome.EnvelopeContent.SignerInfo.SignedAttributes.SigningTime
+	// TODO use authenticSigningTime from signerInfo
+	// https://github.com/notaryproject/notation-core-go/issues/38
+	revocationResult := r.Validate(outcome.EnvelopeContent.SignerInfo.CertificateChain, authenticSigningTime)
+
+	result := &notation.ValidationResult{
+		Type:   trustpolicy.TypeRevocation,
+		Action: outcome.VerificationLevel.Enforcement[trustpolicy.TypeRevocation],
+	}
+
+	// Log all non-nil errors as debug
+	for i, certResult := range revocationResult {
+		for _, err := range certResult.ServerResults {
+			if err != nil {
+				logger.Debugf("error for certificate #%d in chain: %v", (i + 1), err)
+			}
+		}
+	}
+
+	currResult := revocation.OK
+	for _, certResult := range revocationResult {
+		switch certResult.Result {
+		case revocation.OK:
+			continue
+		case revocation.Revoked:
+			currResult = revocation.Revoked
+			// will break out of loop after this
+		default:
+			// revocation.Unknown
+			currResult = revocation.Unknown
+			continue
+		}
+		if currResult == revocation.Revoked {
+			break
+		}
+	}
+
+	switch currResult {
+	case revocation.OK:
+		logger.Debug("no important errors encountered while checking revocation, status is OK")
+	case revocation.Revoked:
+		result.Error = errors.New("one or more certificates in the chain were determined to be revoked, status is Revoked")
+	default:
+		// revocation.Unknown
+		result.Error = errors.New("important error(s) encountered during revocation check, status is Unknown")
```

Comment:
If we are returning error its implicit that it is important.

```suggestion
		result.Error = errors.New("signing certificate(s) revocation status is unknown")
```

Reply:
Resolved in other thread: https://github.com/notaryproject/notation-go/pull/295#discussion_r1163419338